lib/core-server/src/utils/stories-json.ts

```diff
@@ -0,0 +1,65 @@
+import path from 'path';
+import fs from 'fs-extra';
+import glob from 'globby';
+import { logger } from '@storybook/node-logger';
+// import { Options } from '@storybook/core-common';
+import { readCsf } from '@storybook/csf-tools';
+
+interface ExtractedStory {
+  id: string;
+  kind: string;
+  name: string;
+  parameters: Record<string, any>;
+}
+
+type ExtractedStories = Record<string, ExtractedStory>;
+
+export async function extractStoriesJson(fileName: string, storiesGlobs: string[]) {
+  if (!storiesGlobs) {
+    throw new Error('No stories glob');
+  }
+  const storyFiles: string[] = [];
+  await Promise.all(
+    storiesGlobs.map(async (storiesGlob) => {
+      const files = await glob(storiesGlob);
+      storyFiles.push(...files);
+    })
+  );
+  logger.info(`⚙️ Processing ${storyFiles.length} story files`);
+
+  const stories: ExtractedStories = {};
+  await Promise.all(
+    storyFiles.map(async (csfFile) => {
+      const ext = path.extname(csfFile);
+      if (!['.js', '.jsx', '.ts', '.tsx'].includes(ext)) {
+        logger.info(`skipping ${csfFile}`);
+      }
+      const csf = (await readCsf(csfFile)).parse();
+      csf.stories.forEach((story) => {
+        stories[story.id] = { ...story, kind: csf.meta.title };
```

Comment:
Should we be only pulling out the exact `parameters` we want here? Also will the csf tools eventually return other stuff on the story object apart from `parameters`?